usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -313,50 +313,14 @@ def query_elasticsearch(self) -> list:
         sort_field = self.get_elastic_sort_by_fields()
         sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
         record_num = (self.pagination["page"] - 1) * self.pagination["limit"]
-
+        # random page jumping was removed due to performance concerns
         if (self.last_record_sort_value is None and self.last_record_unique_id is not None) or (
             self.last_record_sort_value is not None and self.last_record_unique_id is None
         ):
             # malformed request
             raise Exception(
                 "Using search_after functionality in Elasticsearch requires both last_record_sort_value and last_record_unique_id."
             )
-
-        # API request is asking to jump to a random, non-sequential page of results
-        if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and (
```

Comment:
What is the behavior when an API request is attempting to fetch a random page from ES? Is it handled gracefully?

Reply:
No, it'll just return a 500, which is the same as keyword search currently. I think adding a specific error message would be good, so should I add it to both advanced and keyword?

Reply:
Yes, a 422 with a custom message would be much better. We shouldn't tolerate 500 responses

Reply:
Alright, added 422 to both spending_by_award and spending_by_transaction. Let me know if you have any thoughts on the wording